src/game.cc

```diff
@@ -1404,125 +1481,129 @@ static void showSplash()
-    // Basically just swapping first and last colors, this problem presented ONLY in F2, F1 has right palette in every splash
-    memcpy(palette + (255 * 3), palette, 3);
-    memset(palette, 0, 3);
-
-    for (int i = 0; i < width * height; i++) {
-        if (data[i] == 0) {
-            data[i] = 255;
-        } else if (data[i] == 255) {
-            data[i] = 0;
-        }
-    }
-
-    int size = 0;
-
-    // TODO: Move to settings.
+    int splashScreenStretchMode = 0;
+    int stretchGameMode = 0;
     Config config;
     if (configInit(&config)) {
         if (configRead(&config, "f2_res.ini", false)) {
-            configGetInt(&config, "STATIC_SCREENS", "SPLASH_SCRN_SIZE", &size);
-        }
+            configGetInt(&config, "STATIC_SCREENS", "SPLASH_SCRN_SIZE", &splashScreenStretchMode);
 
+            if (configGetInt(&config, "STATIC_SCREENS", "STRETCH_GAME", &stretchGameMode)) {
+                splashScreenStretchMode = stretchGameMode; // always override if key exists
+            }
+        }
         configFree(&config);
     }
 
     int screenWidth = screenGetWidth();
     int screenHeight = screenGetHeight();
 
-    if (size != 0 || screenWidth < width || screenHeight < height) {
-        int scaledWidth;
-        int scaledHeight;
+    // Stretch image if required or if it doesn't fit the screen
+    if (splashScreenStretchMode != 0 || screenWidth < splashWidth || screenHeight < splashHeight) {
+        int scaledWidth, scaledHeight;
 
-        if (size == 2) {
-            scaledWidth = screenWidth;
-            scaledHeight = screenHeight;
-        } else {
-            if (screenHeight * width >= screenWidth * height) {
-                scaledWidth = screenWidth;
-                scaledHeight = screenWidth * height / width;
-            } else {
-                scaledWidth = screenHeight * width / height;
-                scaledHeight = screenHeight;
-            }
-        }
+        calculateScaledSize(
+            splashWidth,
+            splashHeight,
+            screenWidth,
+            screenHeight,
+            splashScreenStretchMode,
+            scaledWidth,
+            scaledHeight);
 
-        unsigned char* scaled = reinterpret_cast<unsigned char*>(internal_malloc(scaledWidth * scaledHeight));
+        unsigned char* scaled = reinterpret_cast<unsigned char*>(internal_malloc((scaledWidth) * (scaledHeight)));
         if (scaled != nullptr) {
-            blitBufferToBufferStretch(data, width, height, width, scaled, scaledWidth, scaledHeight, scaledWidth);
+            // Stretch the splash screen and fix edge artifacts
+            blitBufferToBufferStretchAndFixEdges(
+                splashData,
+                splashWidth,
+                splashHeight,
+                splashWidth,
+                scaled,
+                scaledWidth,
+                scaledHeight,
+                scaledWidth,
+                1 // numStates = 1 for a single splash image
```

Comment:
We're always using `numStates == 1`… can we remove it?  What's the intended use?